doc/sphinx/practical-tools/utilities.rst

```diff
@@ -190,6 +190,85 @@ since it results in a “packed” plugin: All auxiliary modules (as
 (``Qux_plugin``). This reduces the chances of begin unable to load two
 distinct plugins because of a clash in their auxiliary module names.
 
+.. _arbitrary_arguments:
+
+Passing arbitrary arguments to coqc
++++++++++++++++++++++++++++++++++++
+Using either ``-arg "-w all"`` or ``-arg -w -arg all`` its possible to pass the
+option ``-w all`` to ``coqc``. For options like ``-set 'Default Goal
+Selector=!'`` quotes are necessary: ``-arg "-set 'Default Goal Selector=!'"``.
+It is impossible to pass a double quote using ``-arg``, only single quotes work
+in this situation.
+
+.. _comments:
+
+Comments
+++++++++
+The syntax for comments in a ``_CoqProject`` file is a bit unusual. The
+character ``#`` starts a comment until the end of the line only if it is
+outside a quoted string, and preceded by whitespace or a closing double quote.
+This means ``Filename#.v`` and ``"Filename With # Spaces.v"`` won’t cause
+comments. The ``#`` is in these cases treated like an ordinary character.
+
+.. _grammar_of_coqproject:
+
+The grammar of _CoqProject
+++++++++++++++++++++++++++
+The parser of coq_makefile uses the following grammar:
+
+  .. insertprodn CoqProject unquoted_string
+
+  .. prodn::
+     CoqProject ::= {* {| @blank | @comment | @quoted_string | @unquoted_string } }
+     blank ::= {| space | horizontal_tab | newline }
+     comment ::= # {* comment_byte } newline
+     quoted_string ::= " {* quoted_string_byte } "
+     unquoted_string ::= string_start_byte {* unquoted_string_byte }
+
+Where the following definitions apply:
+
+* :n:`space`, :n:`horizontal_tab` and :n:`newline` stand for the corresponding ASCII values.
+* :n:`comment_byte` is the set of all bytes except :n:`newline`.
+* :n:`quoted_string_byte` is the set of all bytes except ``"``.
+* :n:`string_start_byte` is the set of all bytes except those that match :n:`@blank`, or are ``"`` or ``#``.
+* :n:`unquoted_string_byte` is the set of all bytes except those that match :n:`@blank`.
+
+The parser produces a list of strings in the same order as they were
+encountered in ``_CoqProject``. All :n:`@blank` and :n:`@comment` tokens
+get ignored and the double quotes of :n:`@quoted_string` tokens are removed as
+well.
```

Comment:
That was the reminder I kinda was expecting.
Thanks for the link.

> I expect `unquoted_string` is broken into multiple tokens.

I’m not sure what you mean, but I think not. An `unquoted_string` begins with a `string_start_byte` and goes on until the next `blank`. This whole resulting thing is one token. (That’s why `"` and `#` don’t have any special effects if they occur as `unquoted_string_byte`)

Maybe I can think of some wording that avoids introducing "byte", while still being technically correct. Maybe first pointing out, that _CoqProject files are always treated as if they were UTF-8 encoded, since then distinguishing "byte" from "character" is unnecessary, since all bytes with special semantics are UTF-8 characters and all other bytes UTF-8 encoded text get treated as "ordinary bytes".